cmd/admin-handlers.go

```diff
@@ -572,6 +661,137 @@ func (a adminAPIHandlers) SetConfigHandler(w http.ResponseWriter, r *http.Reques
 	sendServiceCmd(globalAdminPeers, serviceRestart)
 }
 
+func convertValueType(elem []byte, jsonType gjson.Type) (interface{}, error) {
+	str := string(elem)
+	switch jsonType {
+	case gjson.False, gjson.True:
+		return strconv.ParseBool(str)
+	case gjson.JSON:
+		return gjson.Parse(str).Value(), nil
+	case gjson.String:
+		return str, nil
+	case gjson.Number:
+		return strconv.ParseFloat(str, 64)
+	default:
+		return nil, nil
+	}
+}
+
+// SetConfigKeysHandler - PUT /minio/admin/v1/config-keys
+func (a adminAPIHandlers) SetConfigKeysHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "SetConfigKeysHandler")
+
+	// Get current object layer instance.
+	objectAPI := newObjectLayerFn()
+	if objectAPI == nil {
+		writeErrorResponseJSON(w, ErrServerNotInitialized, r.URL)
+		return
+	}
+
+	// Deny if WORM is enabled
+	if globalWORMEnabled {
```

Comment:
This should be allowed if we are toggling worm key, but not for other keys??

Reply:
> This should be allowed if we are toggling worm key, but not for other keys??

Good catch, yeah I agree with you but since this is a delicate topic, let's check with the team if it is okay to change WORM value using SetConfig{,Keys} Admin API

Reply:
No this needs to be handled anyways @vadmeste I don't think anyone else will disagree.

Reply:
> No this needs to be handled anyways @vadmeste I don't think anyone else will disagree.

I kept this code for the moment, meaning we are not going to modify config.json if WORM is enabled in all cases, which should be enough for the moment. What do you think @harshavardhana ?

Reply:
yes @vadmeste it is enough.